app/uk/gov/hmrc/upscanuploadproxy/parsers/ErrorActionParser.scala

```diff
@@ -25,33 +26,42 @@ import play.api.mvc.MultipartFormData.FilePart
 import play.api.mvc.{BodyParser, MultipartFormData, PlayBodyParsers, Result}
 import play.core.parsers.Multipart
 import uk.gov.hmrc.upscanuploadproxy.helpers.Response
+import uk.gov.hmrc.upscanuploadproxy.model.ErrorAction
 
 import scala.concurrent.ExecutionContext
 import scala.util.Try
 
-object RedirectUrlWithKeyParser {
+/*
+ * This is a conventional BodyParser implementation that fails parsing with a Left[Result].
+ * This will be handled by the framework, with the implication that the result will be sent directly to the client.
+ * This is OK here because if we fail to parse and construct the error redirect URL we do not have the option to
+ * redirect the error anyway!
+ */
+object ErrorActionParser {
 
-  def parser(parser: PlayBodyParsers)(implicit ec: ExecutionContext): BodyParser[String] =
+  def parser(parser: PlayBodyParsers)(implicit ec: ExecutionContext): BodyParser[ErrorAction] =
     BodyParser { requestHeader =>
-      parser
-        .multipartFormData(fileIgnoreHandler)(requestHeader)
-        .map(_.right.map(extractRedirectWithKey).joinRight)
+      parser.multipartFormData(fileIgnoreHandler)(requestHeader).map { resultOrMultipartForm =>
```

Comment:
Not keen on the somethingOrSomethingElse variable names. They made it a bit more difficult for me to follow the logic (might just be me). Here I would prefer `_`